fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateDisassemble.java

```diff
@@ -17,144 +17,156 @@
-            }
-            LogicalAggregate mergeAgg = new LogicalAggregate(
-                    groupByExpressionList,
-                    mergeOutputExpressionList,
+            LogicalAggregate globalAggregate = new LogicalAggregate(
+                    globalGroupByExprs,
+                    globalOutputExprs,
                     true,
-                    AggPhase.FIRST_MERGE
+                    AggPhase.GLOBAL
             );
-            return plan(mergeAgg, childPlan);
+            return plan(globalAggregate, plan(localAggregate, plan.child(0)));
         }).toRule(RuleType.AGGREGATE_DISASSEMBLE);
     }
 
-    private org.apache.doris.catalog.AggregateFunction findAggFunc(AggregateFunction functionCall) {
-        FunctionName functionName = new FunctionName(functionCall.getName());
-        List<Expression> expressionList = functionCall.getArguments();
-        List<Type> staleTypeList = expressionList.stream().map(Expression::getDataType)
-                .map(DataType::toCatalogDataType).collect(Collectors.toList());
-        Function staleFuncDesc = new Function(functionName, staleTypeList,
-                functionCall.getDataType().toCatalogDataType(),
-                // I think an aggregate function will never have a variable length parameters
-                false);
-        Function staleFunc = Catalog.getCurrentCatalog()
-                .getFunction(staleFuncDesc, CompareMode.IS_IDENTICAL);
-        Preconditions.checkArgument(staleFunc instanceof org.apache.doris.catalog.AggregateFunction);
-        return  (org.apache.doris.catalog.AggregateFunction) staleFunc;
-    }
+    @SuppressWarnings("InnerClassMayBeStatic")
+    private static class ExpressionReplacer
+            extends ExpressionVisitor<Expression, Map<Expression, Expression>> {
+        private static final ExpressionReplacer INSTANCE = new ExpressionReplacer();
 
-    @SuppressWarnings("unchecked")
-    private <T extends Expression> void replaceSlot(Map<Slot, Slot> staleToNew,
-            List<T> expressionList, Expression root, int index) {
-        if (index != -1) {
-            if (root instanceof Slot) {
-                Slot v = staleToNew.get(root);
-                if (v == null) {
-                    return;
+        @Override
+        public Expression visit(Expression expr, Map<Expression, Expression> substitutionMap) {
+            // TODO: we need to do sub tree match and replace. but we do not have semanticEquals now.
+            //    e.g. a + 1 + 2 in output expression should be replaced by
+            //    (slot reference to update phase out (a + 1)) + 2, if we do group by a + 1
+            //   currently, we could only handle output expression same with group by expression
+            if (substitutionMap.containsKey(expr)) {
+                return substitutionMap.get(expr);
+            } else {
+                List<Expression> newChildren = new ArrayList<>();
```

Comment:
I think you can extends DefaultExpressionRewriter, and replace this line to `return super.visit(expr, substitutionMap)`